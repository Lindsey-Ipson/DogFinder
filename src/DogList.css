.DogList {
  text-align: center;
  align-items: center;
  margin-top: 70px;
}

.DogList img {
  max-width: 200px;
  border-radius: 10px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
}

.DogList-dog-container {
  display: box;
  flex-wrap: wrap;
  justify-content: center;
  margin: 40px;
}

.DogList-name {
  font-size: 1.5rem;
  margin-top: 10px;
}