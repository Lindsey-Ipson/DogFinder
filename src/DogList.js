import React from "react";
import { Link } from "react-router-dom";
import {v1 as uuid} from "uuid";
import "./DogList.css";

const DogList = ({ dogs }) => {
  return (
    <div className='DogList'>
      <h1>Meet the dogs!</h1>
      {dogs.map(dog => (
        <div className="DogList-dog-container" key={uuid()}>
          <img src={dog.src} alt={dog.name} />
          <div className="DogList-name">
            <Link to={`/dogs/${dog.name.toLowerCase()}`}>{dog.name}</Link>
          </div>
        </div>
      ))}
    </div>
  )
};

export default DogList;
