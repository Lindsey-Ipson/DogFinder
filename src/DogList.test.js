import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import DogList from './DogList';

const mockDogs = [
  { name: 'Dog1', src: 'dog1.jpg' },
  { name: 'Dog2', src: 'dog2.jpg' },
];

test('renders DogList component with correct content', () => {
  const { getByText, getAllByAltText, getByRole } = render(
    <Router>
      <DogList dogs={mockDogs} />
    </Router>
  );

  // Check if the title is rendered
  expect(getByText('Meet the dogs!')).toBeInTheDocument();

  // Check if each dog's name is rendered as a link
  mockDogs.forEach((dog) => {
    const link = getByRole('link', { name: new RegExp(dog.name, 'i') });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', `/dogs/${dog.name.toLowerCase()}`);
  });

  // Check if each dog's image is rendered
  mockDogs.forEach((dog) => {
    expect(getAllByAltText(dog.name)).toHaveLength(1);
  });
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <Router>
      <DogList dogs={mockDogs} />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});