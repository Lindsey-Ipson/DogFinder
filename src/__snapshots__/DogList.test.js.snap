// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DogList"
  >
    <h1>
      Meet the dogs!
    </h1>
    <div
      class="DogList-dog-container"
    >
      <img
        alt="Dog1"
        src="dog1.jpg"
      />
      <div
        class="DogList-name"
      >
        <a
          href="/dogs/dog1"
        >
          Dog1
        </a>
      </div>
    </div>
    <div
      class="DogList-dog-container"
    >
      <img
        alt="Dog2"
        src="dog2.jpg"
      />
      <div
        class="DogList-name"
      >
        <a
          href="/dogs/dog2"
        >
          Dog2
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
